styled-switch-view/src/main/java/co/sodalabs/view/StyledSwitch.kt

```diff
@@ -0,0 +1,435 @@
+
+        leftFgArc.set((padding / 10).toFloat(), (padding / 10).toFloat(), ((outerRadii shl 1) - padding / 10).toFloat(), (mHeight - padding / 10).toFloat())
+        rightFgArc.set((mWidth - (outerRadii shl 1) + padding / 10).toFloat(), (padding / 10).toFloat(), (mWidth - padding / 10).toFloat(), (mHeight - padding / 10).toFloat())
+    }
+
+    override fun performClick(): Boolean {
+        super.performClick()
+        if (isOn) {
+            val switchColor = ValueAnimator.ofInt(mWidth - padding - thumbRadii, padding)
+            switchColor.run {
+                addUpdateListener { animation ->
+                    val value = (animation.animatedValue as Int).toFloat()
+                    thumbBounds.set(value, thumbBounds.top, value + thumbRadii, thumbBounds.bottom)
+                    invalidate()
+                }
+                interpolator = AccelerateDecelerateInterpolator()
+                duration = 250
+                start()
+            }
+        } else {
+            val switchColor = ValueAnimator.ofInt(padding, mWidth - padding - thumbRadii)
+            switchColor.run {
+                addUpdateListener { animation ->
+                    val value = (animation.animatedValue as Int).toFloat()
+                    thumbBounds.set(value, thumbBounds.top, value + thumbRadii, thumbBounds.bottom)
+                    invalidate()
+                }
+                interpolator = AccelerateDecelerateInterpolator()
+                duration = 250
+                start()
+            }
+        }
+        isOn = !isOn
+        onToggledListener?.invoke(this, isOn)
+        return true
+    }
+
+    override fun onTouchEvent(event: MotionEvent): Boolean {
+        if (!isEnabled) return false
+
+        val x = event.x
+        when (event.action) {
+            MotionEvent.ACTION_DOWN -> {
+                startTime = System.currentTimeMillis()
+                return true
+            }
+
+            MotionEvent.ACTION_MOVE -> {
+                if (x - thumbRadii.ushr(1) > padding && x + thumbRadii.ushr(1) < mWidth - padding) {
+                    thumbBounds.set(x - thumbRadii.ushr(1), thumbBounds.top, x + thumbRadii.ushr(1), thumbBounds.bottom)
+                    invalidate()
+                }
+                return true
+            }
+
+            MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
+                val endTime = System.currentTimeMillis()
+                val span = endTime - startTime
+                if (span < 200) {
+                    performClick()
+                } else {
+                    if (x >= mWidth.ushr(1)) {
+                        val switchColor = ValueAnimator.ofInt(if (x > mWidth - padding - thumbRadii) mWidth - padding - thumbRadii else x.toInt(), mWidth - padding - thumbRadii)
+                        switchColor.run {
+                            addUpdateListener { animation ->
+                                val value = animation.animatedValue as Float
```

Comment:
There is a casting exception here. Error log is like,

```
    java.lang.ClassCastException: java.lang.Integer cannot be cast to java.lang.Float
        at co.sodalabs.view.StyledSwitch$onTouchEvent$$inlined$run$lambda$2.onAnimationUpdate(StyledSwitch.kt:416)
```

Reply:
It seems like the `animatedValue` is Int and couldn't be cast to Float with `as` operator.